src/web/pages/roles/RoleComponent.tsx

```diff
@@ -45,31 +86,39 @@ const RoleComponent = ({
   const allUsers = useSelector(state => userSelector(state).getAllEntities());
   const allGroups = useSelector(state => groupSelector(state).getAllEntities());
 
+  // @ts-expect-error
   const dispatchLoadAllUsers = () => dispatch(loadAllUsers(gmp)());
+  // @ts-expect-error
   const dispatchLoadAllGroups = () => dispatch(loadAllGroups(gmp)());
 
-  const [dialogVisible, setDialogVisible] = useState(false);
-  const [error, setError] = useState(undefined);
-  const [allPermissions, setAllPermissions] = useState(undefined);
-  const [isCreatingPermission, setIsCreatingPermission] = useState(false);
+  const [dialogVisible, setDialogVisible] = useState<boolean>(false);
+  const [error, setError] = useState<Rejection | undefined>(undefined);
+  const [allPermissions, setAllPermissions] = useState<
+    Array<string> | undefined
+  >(undefined);
+  const [isCreatingPermission, setIsCreatingPermission] =
+    useState<boolean>(false);
   const [isCreatingSuperPermission, setIsCreatingSuperPermission] =
-    useState(false);
-  const [isInUse, setIsInUse] = useState(false);
-  const [isLoadingPermissions, setIsLoadingPermissions] = useState(false);
-  const [permissions, setPermissions] = useState(undefined);
-  const [role, setRole] = useState(undefined);
-  const [title, setTitle] = useState('');
-
-  const openRoleDialog = role => {
+    useState<boolean>(false);
+  const [isInUse, setIsInUse] = useState<boolean>(false);
+  const [isLoadingPermissions, setIsLoadingPermissions] =
+    useState<boolean>(false);
+  const [permissions, setPermissions] = useState<Permission[] | undefined>(
+    undefined,
+  );
+  const [role, setRole] = useState<Role | undefined>(undefined);
+  const [title, setTitle] = useState<string>('');
+
+  const openRoleDialog = (role?: Role) => {
     dispatchLoadAllUsers();
 
-    if (isDefined(role)) {
-      loadSettings(role.id);
+    if (isDefined(role) && role.id) {
+      void loadSettings(role.id);
 
       setDialogVisible(true);
       setIsInUse(role.isInUse());
       setRole(role);
-      setTitle(_('Edit Role {{name}}', role));
+      setTitle(_('Edit Role {{name}}', {name: role.name || ''}));
```

Comment:
In this case the role name is always defined. So i am using the following
```suggestion
      setTitle(_('Edit Role {{name}}', {name: role.name as string}));
```